packages/react/src/components/TextField/TextField.test.tsx

```diff
@@ -527,78 +483,86 @@ describe('TextField with error message', () => {
     jest.runAllTimers();
 
     expect(validator).toHaveBeenCalledTimes(0);
-    assertErrorMessage(wrapper.getDOMNode(), false);
+    assertErrorMessage(container, false);
   });
 
   it('should trigger validation only on focus', () => {
     const validator = jest.fn((value: string) => (value.length > 3 ? errorMessage : ''));
 
-    wrapper = mount(<TextField defaultValue="initial value" onGetErrorMessage={validator} validateOnFocusIn />);
+    const { container } = render(
+      <TextField defaultValue="initial value" onGetErrorMessage={validator} validateOnFocusIn />,
+    );
 
-    const input = wrapper.find('input');
-    input.simulate('input', mockEvent('invalid value'));
+    const input = getInput();
+    userEvent.type(input, 'invalid value', { skipClick: true });
     expect(validator).toHaveBeenCalledTimes(1);
 
-    input.simulate('focus');
+    userEvent.click(input);
     expect(validator).toHaveBeenCalledTimes(2);
 
-    input.simulate('input', mockEvent('also '));
-    input.simulate('input', mockEvent('also invalid'));
-    input.simulate('focus');
+    userEvent.click(container);
+    userEvent.type(input, 'also ');
+    userEvent.type(input, 'also invalid');
     expect(validator).toHaveBeenCalledTimes(3);
   });
 
   it('should trigger validation only on blur', () => {
     const validator = jest.fn((value: string) => (value.length > 3 ? errorMessage : ''));
 
-    wrapper = mount(<TextField defaultValue="initial value" onGetErrorMessage={validator} validateOnFocusOut />);
+    const { container } = render(
+      <TextField defaultValue="initial value" onGetErrorMessage={validator} validateOnFocusOut />,
+    );
 
-    const input = wrapper.find('input');
-    input.simulate('input', mockEvent('invalid value'));
+    const input = getInput();
+    userEvent.type(input, 'invalid value');
     expect(validator).toHaveBeenCalledTimes(1);
 
-    input.simulate('blur');
+    userEvent.click(container);
     expect(validator).toHaveBeenCalledTimes(2);
 
-    input.simulate('input', mockEvent('also '));
-    input.simulate('input', mockEvent('also invalid'));
-
-    input.simulate('blur');
+    userEvent.type(input, 'also ');
+    userEvent.type(input, 'also invalid');
+    userEvent.click(container);
     expect(validator).toHaveBeenCalledTimes(3);
   });
 
   it('should trigger validation on both blur and focus', () => {
     const validator = jest.fn((value: string) => (value.length > 3 ? errorMessage : ''));
 
-    wrapper = mount(
+    const { container } = render(
       <TextField defaultValue="initial value" onGetErrorMessage={validator} validateOnFocusOut validateOnFocusIn />,
     );
 
-    const input = wrapper.find('input');
-    input.simulate('input', mockEvent('value before focus'));
+    const input = getInput();
+    userEvent.type(input, 'value before focus', { skipClick: true });
     expect(validator).toHaveBeenCalledTimes(1);
 
-    input.simulate('focus');
+    userEvent.type(input, 'value after foc ');
```

Comment:
I think this wasn't supposed to have a space after?
```suggestion
    userEvent.type(input, 'value after foc');
```